### PR TITLE
CLI-2589 fix panic when running `api-key store` with resource flag

### PR DESCRIPTION
**Pull request by kevin-wu24**
Release Notes
-------------
Bug Fixes
- Prevents a panic when passing a non-existent Kafka cluster to `confluent api-key store --resource`

Checklist
---------
1. [CRUCIAL] Is the change for features that are already live in prod?
   * yes: ok
   * no: DO NOT MERGE until the required features are live in prod  

What
----
A panic can occur when the user tries to store an api-key secret for a resource that doesn’t exist anymore for that api-key.

References
----------
https://confluentinc.atlassian.net/browse/CLI-2589

### CODE REVIEW

File: internal/cmd/api-key/command_store.go [hunk 1]
```diff
@@ -112,7 +112,7 @@ func (c *command) store(cmd *cobra.Command, args []string) error {
 		return errors.CatchApiKeyForbiddenAccessError(err, getOperation, httpResp)
 	}
 
-	apiKeyIsValidForTargetCluster := (cluster.ID == apiKey.Spec.Resource.Id)
+	apiKeyIsValidForTargetCluster := apiKey.Spec.Resource != nil && cluster.ID == apiKey.Spec.Resource.Id
```
Comment (brianstrauch) [inline internal/cmd/api-key/command_store.go:115]: Can we use panic-safe getter functions here? (or does the SDK not allow it?)
```suggestion
	apiKeyIsValidForTargetCluster := cluster.ID == apiKey.GetSpec().GetResource().GetId()
```

Comment (kevin-wu24) [reply internal/cmd/api-key/command_store.go:115]: The getters are pointer methods, so you could do `apiKey.Spec.Resource.GetID()` which will prevent the same panic as adding the nil check above, so I think I'll do that.

Comment (brianstrauch) [reply internal/cmd/api-key/command_store.go]: (We're still not able to chain multiple methods though, right?)

Comment (kevin-wu24) [reply internal/cmd/api-key/command_store.go]: Unfortunately no, Go doesn't allow for taking the addresses of pointers in a non-roundabout way (i.e. `apiKey.(&GetSpec()).(&GetResource()).GetID().

**Review by brianstrauch — APPROVED** [review]

**Pull request merged**